src/tools/distillery.ml

```diff
@@ -296,16 +296,30 @@ let main () =
     with Not_found -> bad "Not a known template name: %S" s
   in
   let dest_dir = ref None in
-  let check_name name =
-    if not (Str.string_match compilation_unit_name_regexp name 0) then
-      bad "Not a valid compilation unit name: %s" name
+  let check_name =
+    let ret name =
+      if not (Str.string_match compilation_unit_name_regexp name 0) then
+        bad "Not a valid compilation unit name: %s" name
+      else
+        name
+    in
+    fun name ->
+      let name' = String.lowercase name in
```

Comment:
We still support `4.02.x`, and it is a pity to break compatibility just for that.